listenbrainz/listenstore/dump_listenstore.py

```diff
@@ -333,26 +333,33 @@ def write_parquet_files(self,
             }
 
         query = psycopg2.sql.SQL("""
-                     SELECT listened_at,
-                           user_id,
-                           artist_credit_id,
-                           artist_mbids::TEXT[] AS artist_credit_mbids,
-                           artist_credit_name AS m_artist_name,
-                           data->'track_metadata'->>'artist_name' AS l_artist_name,
-                           release_name AS m_release_name,
-                           data->'track_metadata'->>'release_name' AS l_release_name,
-                           release_mbid::TEXT,
-                           recording_name AS m_recording_name,
-                           track_name AS l_recording_name,
-                           mm.recording_mbid::TEXT
-                      FROM listen l
-           FULL OUTER JOIN mbid_mapping mm
-                        ON (data->'track_metadata'->'additional_info'->>'recording_msid')::uuid = mm.recording_msid
-           FULL OUTER JOIN mbid_mapping_metadata m
-                        ON mm.recording_mbid = m.recording_mbid
-                     WHERE {criteria} > %(start)s
-                       AND {criteria} <= %(end)s
-                  ORDER BY {criteria} ASC""").format(criteria=psycopg2.sql.Identifier(criteria))
+        -- can't use coalesce here because we want all listen data or all mapping data to be used for a given
+        -- listen, coalesce mapping data with listen data can yield values from different sources for same listen.
+        -- an alternative is to use to CASE, but need to put case for each column because SQL CASE doesn't allow
+        -- setting multiple columns at once.
+                     SELECT listened_at
+                          , user_id
+                          , artist_credit_id
+                          , artist_mbids::TEXT[] AS m_artist_credit_mbids
+                          , data->'track_metadata'->'additional_info'->>'artist_mbids' AS l_artist_credit_mbids
+                          , artist_credit_name AS m_artist_name
+                          , data->'track_metadata'->>'artist_name' AS l_artist_name
+                          , release_name AS m_release_name
+                          , data->'track_metadata'->>'release_name' AS l_release_name
+                          , release_mbid::TEXT AS m_release_mbid
+                          , data->'track_metadata'->'additional_info'->>'release_mbid' AS l_release_mbid
+                          , recording_name AS m_recording_name
+                          , track_name AS l_recording_name
+                          , mm.recording_mbid::TEXT AS m_recording_mbid
+                          , data->'track_metadata'->'additional_info'->>'recording_mbid' AS l_recording_mbid
+                       FROM listen l
+            FULL OUTER JOIN mbid_mapping mm
```

Comment:
I never got around to making sure we needed full outer join here. Didn't we think that a left join would suffice? Should we do this now that we're mucking with this?

Reply:
Yes, a LEFT JOIN should be enough. will update.